src/WorldIDIdentityManagerImplV1.sol

```diff
@@ -343,6 +359,32 @@ contract WorldIDIdentityManagerImplV1 is OwnableUpgradeable, UUPSUpgradeable, IW
         return _latestRoot;
     }
 
+    function sendRootToStateBridge() public virtual onlyProxy onlyInitialized {
+        if(stateBridgeProxy == address(0x0)) revert stateBridgeHasBeenTurnedOff();        
+
+        (bool success,) = stateBridgeProxy.call(
+            abi.encodeWithSignature("sendRootMultichain(uint256)", _latestRoot)
+        );
+
+        // If the call to the state bridge proxy failed, we revert with a failure.
+        if (!success) revert StateBridgeProxySendRootMultichainFailure();
+    }
+
+    /// @notice Allows a caller to query the address of the current stateBridgeProxy.
+    ///
+    /// @return proxy The address of the currently used stateBridgeProxy
+        function getStateBridgeProxy() public view virtual onlyProxy onlyInitialized returns (address proxy) {
+        return stateBridgeProxy;
+    }
+
+    /// @notice Allows a caller to upgrade the stateBridgeProxy.
+    /// @dev Only the owner of the contract can call this function.
+    ///
+    /// @param _newStateBridgeProxy The address of the new stateBridgeProxy
+    function setStateBridgeProxy(address _newStateBridgeProxy) public virtual onlyProxy onlyInitialized onlyOwner {
```

Comment:
You don't need the `_` here. There's no name clash given you've called it `newStateBridgeProxy`. I would also call the function `setStateBridgeProxyAddress` as it's clearer.

Reply:
done